Add "How to check downloaded file" (#25)
DevExpress/testcafe-examples: examples/check-downloaded-file-name-and-content/README.MD

# Check the Name and the Contents of a Downloaded File

**Test Code**: [index.js](index.js)

> This example shows how to obtain the name and the content of a file downloaded during a test.

Author: edited

> During the test, TestCafe clicks the link that triggers file download. To confirm download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.

Reviewer: ```suggestion
During the test, TestCafe clicks the link that triggers the file download. To confirm download completion and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.
```

> During the test, a link click starts a file download. To verify download success and obtain the name and the content of the file, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.

Reviewer 2: > a link click starts a file download

`a link click starts a file download` sounds like "a cannon shot starts the festival". Maybe it's just `During the test, TestCafe clicks the link that triggers file download.`

> obtain the name and the content of the file

`obtain the file name and content` - much shorter

Author: edited

> During the test, TestCafe clicks the link that triggers file download. To verify download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.

Author: Sound interchangeable to me. We could use `confirm` instead.

> During the test, TestCafe clicks the link that triggers file download. To verify download success and obtain the file name and content, a [RequestLogger](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/) is attached to the test with the [t.requestHooks](https://devexpress.github.io/testcafe/documentation/reference/test-api/test/requesthooks.html) method.

Reviewer 2: Oh, yes, `confirm` is better

> To verify success of a download, a response status code is checked with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method.

Reviewer 3: Not only status code. File name and file content are checked also.

What about this: 

To verify the downloaded file information the [RequestLog.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method is used in combination with [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method.

Author: > The `RequestLog.contains` method in combination with `t.expect.ok` method verify the downloaded file information.

> The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file name and content.

Reviewer: ```suggestion
The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html), verifies the downloaded file name and content.
```

> The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method in combination with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) verify the downloaded file information.

Reviewer 2: What if we say `name and content`? I don't think file content qualifies as `file information` so we may probably be saying it wrong.

Author: Yes, `name and content` is more precise. I edited it in.

> The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html), verifies the downloaded file name and content.

Reviewer: ```suggestion
The [RequestLogger.contains](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/contains.html) method, combined with the [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method, verifies the downloaded file name and content.
```

> The Request Logger is set up to log all successful `GET` requests to a particular url. The responses are then filtered to only contain a `.txt` file with specified contents.

Reviewer 2: URL

Author: edited

> The Request Logger is set up to log all successful `GET` requests to a particular URL. The responses are then filtered to only contain a `.txt` file with specified contents.

Reviewer: ```suggestion
The Request Logger is set up to log all successful `GET` requests to a particular URL. The responses are then filtered to only contain a `.txt` file with the specified content.
```

The Request Logger has the following [options](https://devexpress.github.io/testcafe/documentation/reference/test-api/requestlogger/constructor.html) enabled:

|Option|Effect|
|-|-|
|`logResponseHeaders`| logs the response header|
|`logResponseBody`| logs the response body|
|`stringifyResponseBody`| converts the response body to a string|

> The file name is read from the `content-disposition` response header and is logged to the console. The file contents constitute the response body and are logged to the console.

Reviewer 3: I guess it isn't necessary. To see these logs customer should know about opening the browser devtools.

Author: > The file name is read from the `content-disposition` response header. The file contents constitute the response body.

+removed `.log`s from the test

> The file name is read from the `content-disposition` response header. The file contents constitute the response body.

Reviewer: `The file contents constitute the response body.`

Can you clarify what you mean by "constitute"? For example, can we say:

`The file contents contain the response body.`

Author: I meant to say that "the response body is the file contents". 
If the response body is literally the only thing inside the file, can we still say `The file contents contain the response body` ?

Reviewer: Thank you. Please change it to:

`The response body contains the contents of the file.`

Author: > The response body is added to the file content.
